examples/Azure_IoT_Adu_ESP32/readme.md

```diff
@@ -0,0 +1,272 @@
+  - **Symmetric Key**: follow [this guidance](https://docs.microsoft.com/azure/iot-hub/iot-hub-create-through-portal#register-a-new-device-in-the-iot-hub) to create a device.In this case, the device keys are used to automatically generate a SAS token for authentication.
+  - **X.509 self-signed cert**: Instructions on how to create an X.509 cert for tests can be found [here](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md#configure-and-run-the-samples) (Step 1). Please note that you might need to install some of the [prerequisites](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_linux.md#prerequisites) like OpenSSL.
+- Have the latest [Arduino IDE](https://www.arduino.cc/en/Main/Software) installed.
+
+- Have the [ESP32 board support](https://github.com/espressif/arduino-esp32) installed on Arduino IDE.
+
+  - ESP32 boards are not natively supported by Arduino IDE, so you need to add them manually.
+  - Follow the [instructions](https://github.com/espressif/arduino-esp32) in the official ESP32 repository.
+
+## Setup Instructions
+
+1. Run the Arduino IDE.
+
+1. Install the Azure SDK for Embedded C library.
+
+    - On the Arduino IDE, go to menu `Sketch`, `Include Library`, `Manage Libraries...`.
+    - Search for and install `azure-sdk-for-c`.
+    - **Make sure to install the `1.1.0-beta.1` to use the preview version of the Embedded C SDK.**
+
+1. Open the ESPRESSIF ESP32 sample.
+
+    - On the Arduino IDE, go to menu `File`, `Examples`, `azure-sdk-for-c`.
+    - Click on `Azure_IoT_Adu_ESP32` to open the sample.
+
+1. Configure the ESPRESSIF ESP32 sample.
+
+    Enter your Azure IoT Hub and device information into the sample's `iot_configs.h`:
+    - Add your Wi-Fi SSID to `IOT_CONFIG_WIFI_SSID`
+    - Add your Wi-Fi password to `IOT_CONFIG_WIFI_PASSWORD`
+    - Add you IoTHub Name to `IOT_CONFIG_IOTHUB_FQDN`
+    - Add your Device ID to `IOT_CONFIG_DEVICE_ID`
+    - If using **X.509 Cert**:
+        - Uncomment the `#define IOT_CONFIG_USE_X509_CERT`
+        - Add your cert to `IOT_CONFIG_USE_X509_CERT`
+        - Add your cert PK to `IOT_CONFIG_DEVICE_CERT_PRIVATE_KEY`
+    - If using **Symmetric Key**:
+        - Add your device key to `IOT_CONFIG_DEVICE_KEY`
+    - **IMPORTANT**: make sure to change the `ADU_DEVICE_VERSION` to version 1.1 so that we can build the new update image.
+
+1. Select the appropriate partition scheme for your device. Go to `Tools` -> `Partition Scheme` -> `Minimal SPIFFS`.
+
+## New Image Instructions
+
+In order to update our device, we have to build the image which our device will update to. We will have to direct the Arduino IDE to specify an output directory so that we can easily find the binary. Open the `preferences.txt` located at `C:\Users\<Your User Dir>\AppData\Local\Arduino15\` and add `build.path=C:\Arduino-output` (or whichever directory you prefer).
```

Comment:
We could ask the customer to just get the bin from the temp directory instead of setting a specific destination dir in the config, but I think it's a little more clear doing like how you are suggesting. 

Important: make sure we have a step below mentioning to revert this temp output dir setting.

Btw, in the Arduino IDE build logs this is where it says where the .bin was generated:


```
"C:\\Users\\ewertons\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\esptool_py\\3.1.0/esptool.exe" --chip esp32 elf2image --flash_mode dio --flash_freq 80m --flash_size 4MB -o`
 "C:\\Users\\ewertons\\AppData\\Local\\Temp\\arduino_build_506208/Azure_IoT_Hub_ESP32.ino.bin" 
"C:\\Users\\ewertons\\AppData\\Local\\Temp\\arduino_build_506208/Azure_IoT_Hub_ESP32.ino.elf"
```